notifier/notifiers/http.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2015 clair authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package notifiers implements several kinds of notifier.Notifier
+package notifiers
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/coreos/clair/config"
+	"github.com/coreos/clair/notifier"
+)
+
+// A HTTP notifier dispatches notifications to an HTTP endpoint.
+type HTTP struct {
+	endpoint string
+	client   *http.Client
+}
+
+// A HTTPConfiguration represents the configuration of an HTTP notifier.
+type HTTPConfiguration struct {
+	Endpoint   string
+	ServerName string
+	CertFile   string
+	KeyFile    string
+	CAFile     string
+}
+
+func init() {
+	notifier.RegisterNotifier("http", &HTTP{})
+}
+
+func (h *HTTP) Configure(config *config.NotifierConfig) (bool, error) {
+	// Get configuration
+	var httpConfig HTTPConfiguration
+	if config == nil {
+		return false, nil
+	}
+	if _, ok := config.Params["http"]; !ok {
+		return false, nil
+	}
+	yamlConfig, err := yaml.Marshal(config.Params["http"])
+	if err != nil {
+		return false, errors.New("invalid configuration")
+	}
+	err = yaml.Unmarshal(yamlConfig, &httpConfig)
+	if err != nil {
+		return false, errors.New("invalid configuration")
+	}
```

Comment:
Instead of marshalling and unmarshalling again, couldn't you simply create a type that is `map[string]HTTPConfiguration` and then simply do `httpConfig := parsedYaml["http"]`